src/masternode-sync.cpp

```diff
@@ -83,7 +83,7 @@ void CMasternodeSync::Reset()
 void CMasternodeSync::AddedMasternodeList(uint256 hash)
 {
     if(mnodeman.mapSeenMasternodeBroadcast.count(hash)) {
-        if(mapSeenSyncMNB[hash] < MASTERNODE_SYNC_THRESHOLD) {
+        if(mapSeenSyncMNB.count(hash) && mapSeenSyncMNB[hash] < MASTERNODE_SYNC_THRESHOLD) {
```

Comment:
Hmm, you're right - it should initialize new element rather than crash here... so it probably was protecting us from crashing even more times... hmm..